lasagne/layers/recurrent.py

```diff
@@ -153,61 +177,74 @@ def step_masked(layer_input, mask, hid_previous):
             sequences = input
             step_fun = step
 
-        output = theano.scan(step_fun, sequences=sequences,
-                             go_backwards=self.backwards,
-                             outputs_info=[self.hid_init],
-                             truncate_gradient=self.gradient_steps)[0]
+        # repeat hid_init to batch size
```

Comment:
I'm not sure if that would work with `scan`. You want the output to be of shape `(batchsize, ...)`, so probably you cannot provide a tensor of shape `(1, ...)` in the `outputs_info` argument (even if it is broadcastable in the first dimension). It's worth trying, though. Instead of the dot product, you'd just do `theano.tensor.shape_padleft(self.hid_init, 1)`.